test/extensions/filters/network/thrift_proxy/conn_manager_test.cc

```diff
@@ -1058,6 +1085,178 @@ TEST_F(ThriftConnectionManagerTest, ResetDownstreamConnection) {
   EXPECT_EQ(0U, stats_.request_active_.value());
 }
 
+// Test the base case where there is no limit on the number of requests.
+TEST_F(ThriftConnectionManagerTest, RequestWithNoMaxRequestsLimit) {
+  initializeFilter("");
+  EXPECT_EQ(0, config_->maxRequestsPerConnection());
+
+  EXPECT_EQ(50, sendSomeThriftRequest(50));
+
+  filter_callbacks_.connection_.dispatcher_.clearDeferredDeleteList();
+
+  EXPECT_EQ(0U, store_.counter("test.downstream_cx_max_requests").value());
+  EXPECT_EQ(50U, store_.counter("test.request").value());
+  EXPECT_EQ(50U, store_.counter("test.request_call").value());
+  EXPECT_EQ(0U, stats_.request_active_.value());
+  EXPECT_EQ(50U, store_.counter("test.response").value());
+  EXPECT_EQ(50U, store_.counter("test.response_reply").value());
+  EXPECT_EQ(0U, store_.counter("test.response_exception").value());
+  EXPECT_EQ(0U, store_.counter("test.response_invalid_type").value());
+  EXPECT_EQ(50U, store_.counter("test.response_success").value());
+  EXPECT_EQ(0U, store_.counter("test.response_error").value());
+}
+
+// Test the case where there is a limit on the number of requests but the actual number of requests
+// does not reach the limit.
+TEST_F(ThriftConnectionManagerTest, RequestWithMaxRequestsLimitButNotReach) {
+  const std::string yaml = R"EOF(
+    stat_prefix: test
+    route_config:
+      name: local_route
+    max_requests_per_connection: 50
+    )EOF";
+
+  initializeFilter(yaml);
+  EXPECT_EQ(50, config_->maxRequestsPerConnection());
+
+  EXPECT_EQ(49, sendSomeThriftRequest(49));
```

Comment:
see my comment above, this is counter intuitive... 50 requests should be able to go through if 50 is the configured limit.